### PR TITLE
Route manifest.json files per service

### LINKED ISSUE #1924: Route Manifest.json files for WS Frontpages
**Is your feature request related to a problem? Please describe.**
We currently do have full manifest json files, but they are routed, and used, at `/service/articles/manifest.json`

**Describe the solution you'd like**
Route them at both `/service/articles/manifest.json` and `/service/manifest.json` (dont worry about making it either or, that can be a future optimisation)

Update the current manifest meta tag [here](https://github.com/bbc/simorgh/blob/165cd1643d8c1bcd9ad2889787bb5dc72f17b946/src/app/Layouts/defaultPageWrapper.jsx#L39) to point at `/service/manifest.json` for igbo, yoruba pidgin.

This could be done by loading a `manifestPath=/articles/manifest.json` from service config

**Testing notes**
See dev insight below

Dev insight: Add cypress test to check 200 on manifest URLs, unit tests for routes

If you pull the following into its own container, make a unit test to assert the correct path is used based on the service

```
<Helmet>
  <link rel="manifest" href={`/${service}${manifestPath}`} />
</Helmet>
```

**Additional context**

### PR DESCRIPTION
**Pull request by EinsteinNjoroge**
Resolves #1924 

**Overall change:** _Route frontpages to separate manifest.son files for different services yoruba, pidgin, igbo_

**Code changes:**

- _move manifests link helmet to a separate container_
- _use manifest paths defined in service config. Defaults to  `/service/articles/manifest.json`_

**note**

The diff-coverage will be fixed in [a different issue](https://github.com/bbc/simorgh/issues/1967)

**Testing:**
- The routes `/service/manifest.json` should work
- The routes `/service/articles/manifest.json` should work
- News & Persian Articles should have the manifest in meta tag `/service/articles/manifest.json`
- WS frontpages in storybook should have manifest in meta tag `/service/manifest.json`

---

- [X] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [x] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/Manifest/index.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React, { useContext } from 'react';
+import Helmet from 'react-helmet';
+import { ServiceContext } from '../../contexts/ServiceContext';
+
+const ManifestContainer = () => {
+  const { manifestPath, service } = useContext(ServiceContext);
+  return (
+    <Helmet>
+      {manifestPath ? (
```
Comment (dr3) [inline src/app/containers/Manifest/index.jsx]: 👋 Hey, I think we can require the `manifestPath` to render, maybe do something like

```
const ManifestContainer = () => {
  const { manifestPath, service } = useContext(ServiceContext);
	
  if (!manifestPath) {
	return null;
  }

  return (
    <Helmet>
      <link rel="manifest" href={`/${service}${manifestPath}`} />
    </Helmet>
  );
};
```

Comment (EinsteinNjoroge) [reply src/app/containers/Manifest/index.jsx]: Makes sense.

File: cypress/integration/manifest.js [hunk 2]
```diff
@@ -5,6 +5,10 @@ const testManifest200s = service => {
   it(`should return a 200 status code for ${service}`, () => {
     testResponseCode(`/${service}/articles/manifest.json`, 200);
   });
+
+  it(`should return a 200 status code for ${service} without /articles`, () => {
```
Comment (dr3) [inline cypress/integration/manifest.js:11]: This test sadly wont work as on test and live (in production we go through mozart which arnt routed yet)

| env's | path |
|:------|:-----|
| all   | /news/articles/manifest.json |
| local | /persian/articles/manifest.json |
| local | /igbo/manifest.json |
| local | /yoruba/manifest.json |
| local | /yoruba/manifest.json |

You can limit a test to local only with the `describeForLocalOnly` describe function

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/containers/Manifest/index.test.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    const { href, rel } = linkTags[0];
+
+    expect(linkTags).toHaveLength(1);
+    expect(href).toEqual('/news/manifest.json');
+    expect(rel).toEqual('manifest');
+
+    wrapper.unmount();
+  });
+
+  it('should render with no manifest path provided', () => {
```
Comment (ghost) [inline src/app/containers/Manifest/index.test.jsx]:
```suggestion
  it('should not render with no manifest path provided', () => {
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by dr3 — APPROVED** [review]
LGTM, chosing to ignore code climate as its complaining existing code isnt covered, Ive raised https://github.com/bbc/simorgh/issues/1967 to fix it later, but this will require an admin merge

**Review by ghost — APPROVED** [review]
Looks good.

File: src/app/routes/regex/index.js [hunk 4]
```diff
@@ -18,4 +18,8 @@ export const frontpageRegexPath = `/:service(${serviceRegex}):amp(${ampRegex})?`
 
 export const frontpageDataRegexPath = `${frontpageRegexPath}.json`;
 
+export const manifestRegexPath = `/:service(${serviceRegex})/articles/manifest.json`;
```
Comment (dr3) [inline src/app/routes/regex/index.js]: This is a duplicate of `articleManifestRegexPath` above

File: src/app/routes/regex/index.js [hunk 5]
```diff
@@ -18,4 +18,8 @@ export const frontpageRegexPath = `/:service(${serviceRegex}):amp(${ampRegex})?`
...
+
+export const serviceManifestRegexPath = `/:service(${serviceRegex})/manifest.json`;
```
Comment (dr3) [inline src/app/routes/regex/index.js]: Can you rename this `frontpageManifestRegexPath` to be consistant please :)

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/routes/regex/index.js (hunk 4 above)
Comment (EinsteinNjoroge) [reply src/app/routes/regex/index.js]: @david-boydell Need some help with this

**Review by dr3 — APPROVED** [review]

**Comment by david-boydell** [comment]
All good, this is ready for merge.

**Comment by dr3** [comment]
Needs admin merge due to issue addressed here https://github.com/bbc/simorgh/issues/1967

**Pull request merged**